ComfortZoneApp/app/src/main/java/com/example/comfortzone/WeatherDao.java

```diff
@@ -14,10 +14,12 @@ public interface WeatherDao {
     @Query("SELECT * FROM WeatherData")
     List<WeatherData> getAll();
 
-    // WHERE `temp` BETWEEN :max AND :min
-//    @Query("SELECT * FROM WeatherData WHERE `temp` > :temp")
-//    List<WeatherDao> findByTemp(int temp);
-
     @Insert
     void insertAll(WeatherData... weatherData);
+
+    @Query("DELETE FROM WeatherData")
+    void deleteEntireTable();
+
+    @Query("SELECT * FROM WeatherData WHERE timeUploaded < :hourAgo")
+    List<WeatherData> getUploadTimes(long hourAgo);
```

Comment:
I think the method name is different than what it returns..I wonder if there's different SQL query such that you can simply check and return a boolean if there's one entry that has time uploaded within hr then you can probably just call getAll after, it might be more efficient that way? then you can name this method as `isUploadedWithinHrAgo` or `isWeatherDataFresh` and return boolean